File: Java/src/RotateImageTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class RotateImageTest {

    @Test
    public void testRotate() {
        RotateImage ri = new RotateImage();

        int[][] ints1 = new int[][]{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
        ri.rotate(ints1);
        Assert.assertTrue(Arrays.deepEquals(ints1, new int[][]{{7, 4, 1}, {8, 5, 2}, {9, 6, 3}}));

        int[][] ints2 = new int[][]{{5, 1, 9, 11}, {2, 4, 8, 10}, {13, 3, 6, 7}, {15, 14, 12, 16}};
        ri.rotate(ints2);
        Assert.assertTrue(Arrays.deepEquals(ints2, new int[][]{{15, 13, 2, 5}, {14, 3, 4, 1}, {12, 6, 8, 9}, {16, 7, 10, 11}}));
    }

}

File: Kotlin/src/RotateImageTest.kt
import org.junit.Test

class RotateImageTest {

    @Test
    fun testRotate() {
        val ri = RotateImage()

        val array0 = arrayOf(intArrayOf(1, 2, 3), intArrayOf(4, 5, 6), intArrayOf(7, 8, 9))
        ri.rotate(array0)
        assert(array0.size == 3)
        assert(array0[0].contentEquals(intArrayOf(7, 4, 1)))
        assert(array0[1].contentEquals(intArrayOf(8, 5, 2)))
        assert(array0[2].contentEquals(intArrayOf(9, 6, 3)))

        val array1 = arrayOf(intArrayOf(5, 1, 9, 11), intArrayOf(2, 4, 8, 10), intArrayOf(13, 3, 6, 7), intArrayOf(15, 14, 12, 16))
        ri.rotate(array1)
        assert(array1.size == 4)
        assert(array1[0].contentEquals(intArrayOf(15, 13, 2, 5)))
        assert(array1[1].contentEquals(intArrayOf(14, 3, 4, 1)))
        assert(array1[2].contentEquals(intArrayOf(12, 6, 8, 9)))
        assert(array1[3].contentEquals(intArrayOf(16, 7, 10, 11)))
    }

}